### ISSUE TITLE
Freezing/lagging while dragging programs/windows after 22.04 upgrade

### ISSUE THREAD

**Issue by calvinbull** [pop-os/pop#2317]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

Pop OS 22.04, all application windows.

**Issue/Bug Description:**
Since upgrading to 22.04, my system lags terribly/freezes momentarily while dragging any window. When no windows are being moved, everything runs normally. The issue happens for any window, regardless of the program.

**Steps to reproduce (if you know):**
Open any program and reposition it by dragging. This effect does not change depending on how many programs/windows are open.

**Expected behavior:**
Dragging programs/windows should be smooth and not cause the system to freeze.

**Other Notes:**
- I've reinstalled gnome-tweaks and the couple extensions I'm using but this issue persists with all tweaks/extensions removed.
- This issue was not present on 21.10.

**Comment by leviport** [comment]
Can you describe the hardware you're seeing this on?

**Comment by calvinbull** [comment]
> Can you describe the hardware you're seeing this on?

Here are my hardware specifications via neofetch:
--------------- 
OS: Pop!_OS 22.04 LTS x86_64 
Host: MS-7B93 1.0 
Kernel: 5.17.4-xanmod1 
Uptime: 2 hours, 54 mins 
Shell: bash 5.1.16 
Resolution: 2560x1440 
DE: GNOME 42.0 
WM: Mutter 
WM Theme: Pop 
Theme: Pop-dark [GTK2/3] 
Icons: Pop [GTK2/3] 
Terminal: gnome-terminal 
CPU: AMD Ryzen 7 3800X (16) @ 3.900GH 
GPU: NVIDIA GeForce RTX 2080 Ti Rev.  
Memory: 6084MiB / 32094MiB

**Comment by leviport** [comment]
Does this happen with the default kernel as well?

**Comment by calvinbull** [comment]
> Does this happen with the default kernel as well?

I just removed the 5.17.4-xanmod1 kernel the wrong way - but had a forced lesson in using recovery mode and chroot to install the default kernel again (I see this as a positive, I'm new to linux). 

I have loaded the default 5.16.19-76051619-generic kernel and the problem is nearly completely resolve. I can still get a window to freeze briefly if I drag too quickly, but it is night and day compared to how it was.

Thank you for your suggestion. I'll mark this as closed, but let me know if this is not the correct action as this implies there is a bug with certain kernel versions.
